apps/sharepoint/src/inngest/functions/data-protection/common/helpers.ts

```diff
@@ -56,39 +62,75 @@ export const getChunkedArray = <T>(array: T[], batchSize: number): T[][] => {
   return chunks;
 };
 
-export const formatPermissions = (
-  permission: MicrosoftDriveItemPermission
-): DataProtectionObjectPermission | null => {
-  if (permission.grantedToV2?.user) {
-    return {
-      id: permission.id,
-      type: 'user',
-      displayName: permission.grantedToV2.user.displayName,
-      userId: permission.grantedToV2.user.id,
-      email: permission.grantedToV2.user.email,
-    };
-  } else if (permission.link?.scope === 'anonymous') {
-    return {
-      id: permission.id,
-      type: 'anyone',
-      metadata: {
-        sharedLinks: [permission.link.webUrl],
-      },
-    };
-  }
+export const combinePermisisons = (
+  itemId: string,
+  permissions: MicrosoftDriveItemPermission[]
+): DataProtectionObjectPermission[] => {
+  const combinedArray: CombinedPermission[] = [];
```

Comment:
```suggestion
  const combinedPermissions = new Map<string, CombinedPermission>();
```

Using a map will make function complexity `O(n)` instead of `O(n^2)`.

Reply:
I thought that the array there is not so large, and on small arrays there is almost no difference in speed, but yes, you can use the Map)